django/core/secret_key.py

```diff
@@ -0,0 +1,39 @@
+from django.conf import settings
+from django.utils.module_loading import import_string
+
+
+class SecretKeySettingsBackend:
+    def get_secret_key(self):
+        return settings.SECRET_KEY
+
+    def get_old_keys(self):
+        return list(settings.OLD_SECRET_KEYS)
+
+    def check_secret_key(self):
+
+        from django.core.checks.security.base import (
+            SECRET_KEY_MIN_LENGTH, SECRET_KEY_MIN_UNIQUE_CHARACTERS, W009,
+        )
+
+        passed_check = (
+            getattr(settings, 'SECRET_KEY', None) and
+            len(set(settings.SECRET_KEY)) >= SECRET_KEY_MIN_UNIQUE_CHARACTERS and
+            len(settings.SECRET_KEY) >= SECRET_KEY_MIN_LENGTH
+        )
+        return [] if passed_check else [W009]
```

Comment:
what about checking the old keys?

I think all the check warning and constants should be moved in here too

Reply:
Done, now all keys are checked.